### PR TITLE
[FEAT] Claim Airdrop

### PR DESCRIPTION
**Pull request by adamhannigan**
# Description

This PR introduces the functionality to claim an airdrop that has been sent to your farm. Its first use case will be rewarding the winners of Chicken's community competition.

In the future we will use this mechanism for other rewards that players will receive.

When a player has an airdrop, a treasure chest appears on their farm that wobbles. Once it is claimed, it disappears.

Screen Shot 2022-08-15 at 8 29 56 am
Screen Shot 2022-08-15 at 8 30 00 am

## Type of change

- [X] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Run against the local API.

# Checklist:

- [X] Title of the PR is relevant and is prefixed with [FEAT], [CHORE] or [FIX]
- [X] Screenshot if it includes any UI changes
- [X] I have read the contributing guidelines and agree to the T&Cs
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/styles.css [hunk 1]
```diff
@@ -309,6 +309,38 @@ img {
...
+  40% {
+    transform: translate(3px, 0px);
+  }
+  50% {
+    transform: translate(0px, 0px);
+  }
+  100% {
+    transform: translate(0px, 0px);
+  }
+}
```
Comment (c-py) [inline src/styles.css:342]:
```suggestion
.bulge-repeat {
  animation: bulgeRepeat 1200ms steps(4, jump-none) infinite;
}

@keyframes bulgeRepeat {
  0% {
    transform: scale(1);
  }
  12.5% {
    transform: scale(1.15, 0.9);
  }
  25%, 100% {
    transform: scale(1);
  }
}
```

bulge-reward

File: src/features/game/components/Airdrop.tsx [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+      <img
+        src={chest}
+        className="absolute cursor-pointer hover:img-highlight wobble"
+        onClick={() => setShowModal(true)}
+        style={{
+          width: `${PIXEL_SCALE * 15}px`,
+          left: `${GRID_WIDTH_PX * 59}px`,
+          top: `${GRID_WIDTH_PX * 36}px`,
+        }}
+      />
```
Comment (c-py) [inline src/features/game/components/Airdrop.tsx:90]:
```suggestion
      <div
        className="absolute"
        style={{
          left: `${GRID_WIDTH_PX * 59}px`,
          top: `${GRID_WIDTH_PX * 36}px`,
        }}
      >
        <img src={alerted} className="w-2 mx-3 pb-[12px] animate-float" />
        <img
          src={chest}
          className="absolute cursor-pointer hover:img-highlight bulge-repeat"
          onClick={() => setShowModal(true)}
          style={{
            width: `${PIXEL_SCALE * 15}px`,
          }}
        />
      </div>
```

File: src/features/game/components/Airdrop.tsx [hunk 3]
```diff
@@ -0,0 +1,93 @@
+import { useActor } from "@xstate/react";
+import { Button } from "components/ui/Button";
+import { Panel } from "components/ui/Panel";
+import React, { useContext, useState } from "react";
+import { Modal } from "react-bootstrap";
+
+import chest from "src/assets/icons/chest.png";
+import token from "src/assets/icons/token.png";
```
Comment (c-py) [inline src/features/game/components/Airdrop.tsx:8]:
```suggestion
import chest from "src/assets/icons/chest.png";
import token from "src/assets/icons/token.png";
import alerted from "assets/icons/expression_alerted.png";
```

**Review by c-py — CHANGES_REQUESTED** [review]
**suggestion (blocking):** I think the wobble animation is a little unnatural. It's a bit too fluid, and makes the terrain seems icy.

I have made a suggestion using the bulge animation. What do you think?

bulge-reward

File: src/features/game/components/Airdrop.tsx [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+            </div>
+          )}
+
+          {itemNames.length > 0 &&
+            itemNames.map((name) => (
+              <div className="flex justify-end items-center mb-1" key={name}>
+                <img src={ITEM_DETAILS[name].image} className="w-6 mr-2" />
+                <span>{airdrop.items[name]}</span>
+              </div>
+            ))}
```
Comment (c-py) [inline src/features/game/components/Airdrop.tsx:73]: **suggestion (non-blocking):** Perhaps the modal could be adjusted to have the full resource names. There is quite a bit of empty space on the left hand side.

I've had a go at trying to fill out the space. I don't think my attempt looks very good, but here it is anyway 😁 

modal


```suggestion
<span>Congratulations, you found a reward!</span>

          <div className="flex flex-col pt-4">
            {airdrop.sfl && (
              <div className="flex items-center justify-center mb-2">
                <img src={token} className="w-6 mr-2" />
                <span>{airdrop.sfl} SFL</span>
              </div>
            )}

            {itemNames.length > 0 &&
              itemNames.map((name) => (
                <div
                  className="flex items-center justify-center mb-2"
                  key={name}
                >
                  <img src={ITEM_DETAILS[name].image} className="w-6 mr-2" />
                  <span>
                    {airdrop.items[name]} {name}
                  </span>
                </div>
              ))}
          </div>
```

File: src/features/game/components/Airdrop.tsx (hunk 2 above)
Comment (adamhannigan) [reply src/features/game/components/Airdrop.tsx:90]: Agreed! Impressive suggestion and thanks for going above and beyond to commit it in! 🏆

File: src/features/game/components/Airdrop.tsx (hunk 4 above)
Comment (adamhannigan) [reply src/features/game/components/Airdrop.tsx:74]: I agree this looks better.

@spencerdezartsmith Perhaps we can circle back on some standards for alignment of text in modals

**Review by c-py — APPROVED** [review]
LGTM!

**Review by c-py — APPROVED** [review]
LGTM!

**Pull request merged**